src/OrbitBase/ThreadPoolTest.cpp

```diff
@@ -378,3 +380,120 @@ TEST(ThreadPool, WaitWithoutShutdown) {
       },
       "");
 }
+
+TEST(ThreadPool, FutureBasic) {
+  constexpr size_t kThreadPoolMinSize = 1;
+  constexpr size_t kThreadPoolMaxSize = 2;
+  constexpr absl::Duration kThreadTtl = absl::Milliseconds(5);
+  std::unique_ptr<ThreadPool> thread_pool =
+      ThreadPool::Create(kThreadPoolMinSize, kThreadPoolMaxSize, kThreadTtl);
+
+  absl::Mutex mutex;
+  bool called = false;
+
+  {
+    absl::MutexLock lock(&mutex);
+    orbit_base::Future<void> future = thread_pool->Schedule([&]() {
+      absl::MutexLock lock(&mutex);
+      called = true;
+    });
+
+    EXPECT_TRUE(future.IsValid());
+    EXPECT_FALSE(future.IsFinished());
+
+    EXPECT_TRUE(mutex.AwaitWithTimeout(absl::Condition(
+                                           +[](bool* called) { return *called; }, &called),
+                                       absl::Milliseconds(100)));
+
+    for (int elapsed_ms = 0; !future.IsFinished() && elapsed_ms < 100; ++elapsed_ms) {
+      absl::SleepFor(absl::Milliseconds(1));
+    }
```

Comment:
What is the purpose of this loop? At this point we know the lambda has been executed because it released the mutex. We are waiting for the future to be marked as finished. Why not call future.get()? Isn't there a chance that the marking as finished take more than 100ms?

Reply:
Yes, this part is not ideal. But this will change with your suggested behaviour change of Get(). Thanks for raising!

Reply:
Done.